sys-fs/squashfs_tools/patches/squashfs_tools-4.3-gcc2.patch

```diff
@@ -0,0 +1,50 @@
+--- a/unsquash-2.c
++++ b/unsquash-2.c
+@@ -117,11 +117,11 @@ int read_fragment_table_2(long long *directory_table_end)
+ 
+ void read_fragment_2(unsigned int fragment, long long *start_block, int *size)
+ {
+-	TRACE("read_fragment: reading fragment %d\n", fragment);
+-
+ 	squashfs_fragment_entry_2 *fragment_entry = &fragment_table[fragment];
+ 	*start_block = fragment_entry->start_block;
+ 	*size = fragment_entry->size;
++
++	TRACE("read_fragment: reading fragment %d\n", fragment);
```

Comment:
I've just generated a patchset with haikuporter. It starts with a mail-style header and then contains the patch:
```Diff
diff --git a/unsquash-2.c b/unsquash-2.c
index 0f2bfe8..5a3c39b 100644
--- a/unsquash-2.c
+++ b/unsquash-2.c
@@ -118,13 +118,13 @@ int read_fragment_table_2(long long *directory_table_end)
 void read_fragment_2(unsigned int fragment, long long *start_block, int *size)
 {
 	TRACE("read_fragment: reading fragment %d\n", fragment);
-
+	{
 	squashfs_fragment_entry_2 *fragment_entry = &fragment_table[fragment];
 	*start_block = fragment_entry->start_block;
 	*size = fragment_entry->size;
+	}
 }
 
-
 struct inode *read_inode_2(unsigned int start_block, unsigned int offset)
 {
 	static union squashfs_inode_header_2 header;
diff --git a/unsquash-3.c b/unsquash-3.c
index d5af57a..b5177c0 100644
--- a/unsquash-3.c
+++ b/unsquash-3.c
@@ -105,13 +105,13 @@ int read_fragment_table_3(long long *directory_table_end)
 void read_fragment_3(unsigned int fragment, long long *start_block, int *size)
 {
 	TRACE("read_fragment: reading fragment %d\n", fragment);
-
+	{
 	squashfs_fragment_entry_3 *fragment_entry = &fragment_table[fragment];
 	*start_block = fragment_entry->start_block;
 	*size = fragment_entry->size;
+	}
 }
 
-
 struct inode *read_inode_3(unsigned int start_block, unsigned int offset)
 {
 	static union squashfs_inode_header_3 header;
```

In the beginning it's a bit hard to understand what we need to do to have haikuporter generate pathsets.
But basically, if you start from scratch without any patch, you:
```
hp -b squashfs_tools
cd work-4.3/sources/squashfs4.3
vi unsquash-2.c unsquash-3.c
git add unsquash-2.c unsquash-3.c
git commit -m "gcc2 patch" unsquash-2.c unsquash-3.c
cd ../../..
hp -e squashfs_tools
```
(hp is an alias to "haikuporter --get-dependencies")
Warning: "hp -e" will delete any file called squashfs_tools-4.3.patch and overwrite squashfs_tools-4.3.patchset without any notice, so don't forget to make backups ;-)

Now if you already have a patchset with several distinct patches, you can change any of them by adding a commit and then doing an interactive squash/rebase against the git revision that corresponds to the import.
I can detail that if you need.

The other difficult topic is when you have a patch for an old release, and it does not apply to the new release. It took me a while to become confortable with how to use "git add" and "git am --continue" to refresh a patchset. If you need more info I can explain what I have learned ;-)

Using patchsets is recommended, especially if the patch is not trivial, because it simplifies the maintainance of the patchsets when we upgrade a recipe.